sui_core/src/gateway_types.rs

```diff
@@ -0,0 +1,947 @@
+        writeln!(writer, "Transaction Signature: {:?}", self.tx_signature)?;
+        writeln!(
+            writer,
+            "Signed Authorities : {:?}",
+            self.auth_sign_info
+                .signatures
+                .iter()
+                .map(|(name, _)| name)
+                .collect::<Vec<_>>()
+        )?;
+        write!(writer, "{}", &self.data)?;
+        write!(f, "{}", writer)
+    }
+}
+
+impl TryFrom<CertifiedTransaction> for SuiCertifiedTransaction {
+    type Error = anyhow::Error;
+
+    fn try_from(cert: CertifiedTransaction) -> Result<Self, Self::Error> {
+        Ok(Self {
+            transaction_digest: *cert.digest(),
+            data: cert.data.try_into()?,
+            tx_signature: cert.tx_signature,
+            auth_sign_info: cert.auth_sign_info,
+        })
+    }
+}
+
+/// The response from processing a transaction or a certified transaction
+#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize, JsonSchema)]
+#[serde(rename = "TransactionEffects", rename_all = "camelCase")]
+pub struct SuiTransactionEffects {
+    // The status of the execution
+    pub status: SuiExecutionStatus,
+    // The object references of the shared objects used in this transaction. Empty if no shared objects were used.
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub shared_objects: Vec<SuiObjectRef>,
+    // The transaction digest
+    pub transaction_digest: TransactionDigest,
+    // ObjectRef and owner of new objects created.
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub created: Vec<OwnedObjectRef>,
+    // ObjectRef and owner of mutated objects, including gas object.
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub mutated: Vec<OwnedObjectRef>,
+    // ObjectRef and owner of objects that are unwrapped in this transaction.
+    // Unwrapped objects are objects that were wrapped into other objects in the past,
+    // and just got extracted out.
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub unwrapped: Vec<OwnedObjectRef>,
+    // Object Refs of objects now deleted (the old refs).
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub deleted: Vec<SuiObjectRef>,
+    // Object refs of objects now wrapped in other objects.
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub wrapped: Vec<SuiObjectRef>,
+    // The updated gas object reference. Have a dedicated field for convenient access.
+    // It's also included in mutated.
+    pub gas_object: OwnedObjectRef,
+    /// The events emitted during execution. Note that only successful transactions emit events
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub events: Vec<SuiEvent>,
+    /// The set of transaction digests this transaction depends on.
+    #[serde(default, skip_serializing_if = "Vec::is_empty")]
+    pub dependencies: Vec<String>,
```

Comment:
nit: should this be `SuiTransactionEffects`?
```suggestion
    pub dependencies: Vec<TransactionDigest>,
```